components/TextFieldMultiple.js

```diff
@@ -0,0 +1,189 @@
+import React, { Component } from 'react';
+import {
+  View,
+  Text,
+  StyleSheet,
+  ViewPropTypes,
+} from 'react-native';
+
+import { Icon, Input } from 'native-base';
+import { Col, Row } from 'react-native-easy-grid';
+import PropTypes from 'prop-types';
+
+const styles = StyleSheet.create({
+
+  removeIcon: {
+    fontSize: 30,
+    color: 'black',
+    paddingLeft: 10,
+    paddingRight: 10,
+    paddingTop: 10,
+  },
+  containerStyle: {
+    alignSelf: 'stretch',
+    marginVertical: 10,
+    padding: 5,
+    alignItems: 'flex-start',
+  },
+  formIconLabel: {
+    width: 'auto',
+  },
+  input: {
+    marginBottom: 50,
+    marginLeft: 25,
+  },
+  textInputStyle: {
+    borderBottomWidth: 1,
+    borderStyle: 'solid',
+    borderColor: '#D9D5DC',
+  },
+});
+
+
+class TextFieldMultiple extends Component {
+  constructor(props) {
+    super(props);
+    const { items } = this.props;
+    const existValues = (items || []).filter(value => !value.delete);
+    if (!existValues || !existValues.length || existValues[existValues.length - 1].value) {
+      items.push({
+        value: '',
+      });
+    }
+    this.state = {
+      values: props.items,
+    };
+  }
+
+  onFieldChange(value, index) {
+    const { values } = this.state;
+    let isLast = true;
+    for (let i = index + 1; i < values.length; i++) {
+      isLast = values[i].delete;
+      if (!isLast) {
+        break;
+      }
+    }
+    this.setState((prevState) => {
+      const { values: prevValues } = prevState;
+      prevValues[index].value = value;
+
+      if (isLast && value.trim()) {
+        prevValues.push({
+          value: '',
+        });
+      }
+      return {
+        prevValues,
+      };
+    });
+    this.props.onChange(values);
```

Comment:
`values` will not contain the value just added (`value`). It would probably be best to pass `prevValues` into the `onChange` function, so we might need to move this into the `setState` function so that it is within the scope where `prevValues` exists.